### ISSUE TITLE
[css-font-loading] [info discovery] iterable doesn't tell you what index you are

### ISSUE THREAD

**Issue by litherum** [w3c/csswg-drafts#6593]
One natural use case use case of the information discovery facilities in css-font-loading would be "I want to swap the green color in my palette to my custom color." Doing this requires 2 steps:

1. Iterating over the colors in the palette until you find green
2. Creating a `@font-palette-values` rule that maps the index of green to your new custom color

The trick here is "index" of green. When you iterate through an `iterable`, it doesn't tell you what index you're at. So, if you want to determine what index green lies at, you have to track that yourself.

You can do this with a 3-clause for loop, and `getter`, but maybe we can do better here.

**Comment by tabatkins** [comment]
While it's not the default iterator, you can iterate over the `.entries()` iterator to get the index and the value at the same time. All three of the iterator methods are provided by `iterable<>`.

**Comment by litherum** [comment]
Does `entries()` actually produce the index and the value at the same time?

**Comment by tabatkins** [comment]
Yeah, `.keys()` produces an iterator for the keys (numeric indexes in this case), `.values()` (the default for arrays) produces just the values, and `.entries()` produces `[index, value]` pairs.

You can then conveniently destructure over them:

```js
for(let [i, v] of palette.entries()) { ... }
```

**Comment by litherum** [comment]
https://heycam.github.io/webidl/#idl-iterable

> If a single type parameter is given, then the interface has a value iterator and provides values of the specified type. If two type parameters are given, then the interface has a pair iterator and provides value pairs with the given types.

Currently, the spec just has `iterable<DOMString>`, not `iterable<unsigned long, DOMString>`. Maybe I'm misunderstanding?

**Comment by tabatkins** [comment]
That's describing the default iterators, yes. Look slightly further down:

> Interfaces with iterable declarations must not have any interface members named "entries", "forEach", "keys", or "values", or have any inherited interfaces that have members with these names.

This is because those four methods are automatically provided by the iterable declaration (with the specifics defined in <https://heycam.github.io/webidl/#es-iterable>)
